frontend/src/app/organization/organization-admin/organization-admin.component.ts

```diff
@@ -36,25 +35,19 @@ export class OrganizationAdminComponent {
   public static Route = {
     path: 'admin',
     component: OrganizationAdminComponent,
-    title: 'Organization Administration',
-    resolve: { profile: profileResolver }
+    title: 'Organization Administration'
   };
 
   constructor(
-    private route: ActivatedRoute,
     private router: Router,
     private snackBar: MatSnackBar,
+    private profileService: ProfileService,
     private organizationService: OrganizationService,
     private permissionService: PermissionService
   ) {
+    this.profile = this.profileService.profile()!;
```

Comment:
The use of `!` here feels unsatisfying. Opening this route incognito leads to an error in console of trying to read permissions of undefined.

Perhaps this page should be protected by a guard of some sort. The other thing this points out to me, though, is perhaps there's a better way of ensuring that for pages we expect a user to be logged in on, we find a cleaner, idiomatic way to get that context into the components and can avoid this kind of coercion.